### PR TITLE
Adding metadata labels test

### PR DESCRIPTION
**Pull request by datwiz**
Provides feature for Issue #102.
Follows the example of EnvVar tests for Labels.

* Updated README.md to include Labels under Metadata Tests 
* Added Labels to the imageConfig and drivers.
* Added Label key and value checking.  Value checks support regexes.
* Modified the EnvVar value checking to also support regexes.  (This may or may not be a desirable change.)
* The key presence checking has also been modified to support env vars and labels with empty string values.  Prior, labels or env vars with empty values would result in the label key check failing.
* Added an optional test case for image metadata testing, but currently skipped in the test driver script.  This new test builds a new image, based on debian8, to add env vars and labels.  Wasn't sure if this was desired behaviour or not.

### CODE REVIEW

**Comment by datwiz** [comment]
All changes in the pull request are mine, but did use a different email address for some commits.

File: pkg/drivers/driver.go [hunk 1]
```diff
@@ -79,3 +79,13 @@ func convertEnvToMap(env []string) map[string]string {
 	}
 	return envMap
 }
+
+func convertLabelsToMap(labels []string) map[string]string {
```
Comment (nkubala) [inline pkg/drivers/driver.go]: this logic is identical to `convertEnvToMap`, can you just rename that function to `convertSliceToMap` or something and reuse that here?

**Comment by nkubala** [comment]
Can you also go ahead and squash your commits and amend the commit message to use the email address you used to sign the CLA?

File: tests/structure_test_tests.sh [hunk 2]
```diff
@@ -50,7 +68,7 @@ then
   failures=$((failures +1))
 fi
 
-# Test the image.
+Test the image.
```
[the remark this answers is not in the document]
Comment (datwiz) [reply tests/structure_test_tests.sh]: Yes, it should

File: pkg/drivers/driver.go (hunk 1 above)
Comment (datwiz) [reply pkg/drivers/driver.go]: can do

**Comment by datwiz** [comment]
Will address the comments.
Question:  The image metadata test case in `structure_test_tests.sh` is currently skipped as the test case requires building a new image with expected metadata added.  Before removing the skip flag, I wanted to check if there were any issues with having test cases that build new images and whether the image needs to be removed at the conclusion of the test or if Travis cleans up automatically at the end of the job.
Also happy to add more container tests for further config metadata.  Do you have an ETA for when the container-diff code will get added to vendor here?

**Comment by datwiz** [comment]
I’ve also since added my new email address to my google account and the CLA checker looks happy.  Did you still want the commits squashed?

**Comment by tejal29** [comment]
LGTM! No need to squash the commits. The merge to master should do it.

**Comment by nkubala** [comment]
> I wanted to check if there were any issues with having test cases that build new images and whether the image needs to be removed at the conclusion of the test or if Travis cleans up automatically at the end of the job.

I think this should be fine, it does mean that we'll need Docker wherever we run these tests but since we're already mounting in a `docker.sock` for our bazel test run we're not requiring anything new there.

> Do you have an ETA for when the container-diff code will get added to vendor here?

I'll try and do it today

**Comment by datwiz** [comment]
pushed fixes and have enabled the image metadata tests.  Appears to have run the tests successfully.

**Pull request merged**